### DISCUSSION TITLE
replace unique_ptr constructor with `MakeUnique` or `UniqueIterator`

### DISCUSSION THREAD

**Discussion by gogim1** [apache/kvrocks#719] [General]
ref https://github.com/apache/incubator-kvrocks/pull/711#discussion_r913359020
ref https://github.com/apache/incubator-kvrocks/pull/703#discussion_r912320795

I noticed there are many places in the code base where used the unique_ptr constructor directly instead of using `MakeUnique` and `UniqueIterator`. Do we need to replace them all? I wonder if this fix is worth PRs as it is too trivial and doesn't seem to bring performance benefits.

**Comment by PragmaTwice** [comment]
I think it is OK to replace them to keep a unified code style.
